### ISSUE TITLE
Plugin attempts to validate API keys when the Sezzle payment method is disabled

### ISSUE THREAD

**Issue by cbhdalton** [sezzle/sezzle-magento2#24]
### Issue
There appears to be a bug in the `Plugin/Config/Controller/Adminhtml/System/Config/SavePlugin.php` file related to validating the API keys.

In the `aroundExecute` method, there's a section that appears to try to validate the Sezzle API keys when they've changed:

```php
            // only validate keys if they are changed
            if ($this->hasKeysChanged($oldConfig, $newConfig)) {
                if (!$merchantUUID = $this->authenticationService->validateAPIKeys(
                    $newConfig[SezzleConfig::KEY_PUBLIC_KEY],
                    $newConfig[SezzleConfig::KEY_PRIVATE_KEY],
                    $newConfig[SezzleConfig::KEY_PAYMENT_MODE]
                )) {
                    throw new ValidationException(__('Auth token not found.'));
                }
            }
``` 

The core issue is there's no way to leave empty or invalid keys in the config when you try to save the configurations on the payment methods, effectively locking store owners out of modifying their payment methods without first setting up valid Sezzle API keys. This can cause headaches with the store owners if they need to change other payment settings before they've gotten their API keys, or if they're running multiple stores and only want the payment method enabled on one of the stores.

### Steps to Replicate

1. Clone the repo and follow the setup instructions. (You may need to update the database image in the docker-compose.yml file too. `mysql:5.6.23` seems to use an older docker scheme that didn't work for me. Updating it to `mysql:5.6.51` fixed the setup issue for me.)
2. Navigate to the payment method settings - Store > Config > Sales > Payment Methods
3. Try to save the page without changing anything. You should receive an error message saying "Unable to validate the Sezzle API Keys." and any changes you've made won't save.

![Screenshot from 2024-08-07 09-34-40](https://github.com/user-attachments/assets/2848f922-d625-415d-ae63-73c27483510d)

### Suggested Change(s)
I've got two ideas for fixing this issue that seem to work for me:

- Add an additional check to only validate the keys when they've changed **AND** when the payment method is enabled in the config. This seems like the most logical since there shouldn't really be any reason to check if API keys are working if the store owner doesn't even have the payment method enabled. Just changing the `if` statement above to `$newConfig['sezzle_enabled'] && $this->hasKeysChanged($oldConfig, $newConfig)` would do the trick.
- Modify the logic of the `hasKeysChanged` method to use the inequality operator `!=` rather than the non-identity operator `!==`. The reason the extension tests the API keys when they're empty even when they've not really changed is because the key data in the post request is an empty string, while the data in the database is `null` and the non-identity operator will see these as different values thus triggering the API key validation logic. This fixes the issue, but doesn't make as much sense to do since it only really covers the case where the store owner doesn't have them set in the first place, and presumably they'd still have the payment method disabled anyway.

**Comment by arijit-sezzle** [comment]
We are facing issues in releasing the version in the Magento marketplace. We will update here once the new version is released. Thanks.

**Comment by arijit-sezzle** [comment]
Changes released in 7.0.17
